lib/mongo_compat_api.js

```diff
@@ -0,0 +1,325 @@
+ * both, shared and dedicated database.
+ *
+ * @param ditcher ditcher instance
+ * @param params default params for the current app
+ * @constructor
+ */
+var MongoCompatApi = function (ditcher, params) {
+  assert.ok(params.__fhdb, "__fhdb parameter is required");
+  assert.ok(typeof params.__dbperapp !== 'undefined', "__dbperapp parameter is required");
+
+  if(params.__dbperapp === false) {
+    /**
+     * Copy the default options that are always required by ditcher
+     * @param target The options object to copy them to
+     * @returns The options object plus the default properties
+     */
+    this.copyDefaultOptions = function (target) {
+      target.__fhdb = params.__fhdb;
+      target.__dbperapp = params.__dbperapp;
+      return target;
+    };
+
+    /**
+     * Create a new collection
+     * http://mongodb.github.io/node-mongodb-native/2.0/api/Db.html#createCollection
+     * @param name collection name
+     * @param (optional) options
+     * @param (optional) callback results callback
+     * @returns Promise if no callback passed
+     */
+    this.createCollection = function (name, options, callback) {
+      assert.ok(name, "collection name required");
+
+      if (typeof options === "function") {
+        callback = options;
+        options = {};
+      }
+
+      options = this.copyDefaultOptions(options || {});
+      options.type = name;
+
+      if (callback) {
+        ditcher.doCreate(options, callback);
+      } else {
+        return new Promise(function (resolve, reject) {
+          ditcher.doCreate(options, function (err, collection) {
+            if (err) {
+              return reject(err);
+            }
+
+            resolve(collection);
+          });
+        });
+      }
+    };
+
+    /**
+     * Drop a collection
+     * http://mongodb.github.io/node-mongodb-native/2.0/api/Db.html#dropCollection
+     * @param name
+     * @param (optional) callback
+     * @returns Promise if no callback passed
+     */
+    this.dropCollection = function (name, callback) {
+      assert.ok(name, "collection name required");
+
+      var options = {
+        type: name
+      };
+
+      options = this.copyDefaultOptions(options || {});
+
+      if (callback) {
+        ditcher.doDropCollection(options, callback);
+      } else {
+        return new Promise(function (resolve, reject) {
+          ditcher.doDropCollection(options, function (err, result) {
+            if (err) {
+              return reject(err);
+            }
+
+            resolve(result);
+          });
+        });
+      }
+    };
+
+    /**
+     * List all collections
+     * http://mongodb.github.io/node-mongodb-native/2.0/api/Db.html#listCollections
+     * @param filter Query to filter collections by
+     * @param options (optional) Optional settings
+     * @returns Cursor
+     */
+    this.listCollections = function (filter, options) {
+      var options = this.copyDefaultOptions(options || {});
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "options" as this name is already used in declaration at line 100. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2814)